src/components/modal/types.ts

```diff
@@ -1,10 +1,33 @@
 import { ForwardRefExoticComponent, HTMLAttributes, RefAttributes } from 'react'
 
+// Define an enum-like type for the possible variants of the modal
+export type ModalVariant = 'small' | 'medium'
 export interface ModalProps extends HTMLAttributes<HTMLDivElement> {
```

Comment:
**question:** Are you planning further changes to this interface in a future PR?

**suggestion:** I think it would be ideal to aim for an API like:
```javascript
<Dialog size="small" onClose={...}>
  <Dialog.Header>Title text</Dialog.Header>
  <Dialog.Body>
    ...
  </Dialog.Body>
  <Dialog.Footer>
    <ButtonGroup>
      <Button ...>Button 1</Button>
      <Button ...>Button 2</Button>
    </ButtonGroup>
  </Dialog.Footer>
</Dialog>
```

This affords us (and the Design System) a lot of flexibility to evolve over time as it makes it trivial to achieve new layouts within a dialog. When we hardcode the use of atoms like ModalFooter and ModalBody within the Modal atom, we significantly constrain ourselves.

**note:** The naming of some of the atoms above could be discussed with Design (e.g. Header vs Title 🤷).

Reply:
yes, and that's what I want to achieve in the next PR